### PR TITLE
Coding Fhir type added to tokenIndex()

### LINKED ISSUE #1607: Make sure all the data types for particular Search Parameter Types are indexed properly in the engine database.
A related issue to this is that Encounter.CLASS search does not work. This is because fhirType "Coding" is missing from the tokenIndex function in ResourceIndexer.kt. We need to add these lines.

```
 "Coding" -> {
        val coding = value as ICoding
        listOf(TokenIndex(searchParam.name, searchParam.path, coding.system ?: "", coding.code))
      }
```

cc: @aditya-07

_Originally posted by @joiskash in https://github.com/google/android-fhir/issues/1306#issuecomment-1237711713_

See https://www.hl7.org/fhir/search.html#ptypes for the list of data types to be handled per search parameter type.

### PR DESCRIPTION
**Pull request by PallaviGanorkar**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1607 

**Description**
Coding Fhir type added to tokenIndex() function for search

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by PallaviGanorkar** [comment]
@joiskash Please do first round of review.

**Review by joiskash — APPROVED** [review]
Looks good to me!

**Comment by PallaviGanorkar** [comment]
@aditya-07  Could you please do second round of review?

File: engine/src/test/java/com/google/android/fhir/index/ResourceIndexerTest.kt [hunk 1]
```diff
@@ -556,6 +557,26 @@ class ResourceIndexerTest {
       .contains(TokenIndex("code", "Observation.code", systemString, codeString))
   }
 
+  @Test
+  fun index_token_Coding() {
+    val codeString = "1427AAAAA"
+    val systemString = "http://openmrs.org/concepts"
```
Comment (aditya-07) [inline engine/src/test/java/com/google/android/fhir/index/ResourceIndexerTest.kt:563]: Remove the variables and use the values directly in both the places.

**Review by aditya-07 — APPROVED** [review]
Left a small comment, otherwise LGTM !

**Pull request merged**